src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -25,6 +35,70 @@ static void EnsureLibSsl10Initialized()
 }
 #endif
 
+static int32_t g_config_specified_ciphersuites = 0;
+
+static void DetectCiphersuiteConfiguration()
+{
+    // This routine will always produce g_config_specified_ciphersuites = 0 on OpenSSL 1.0.x,
+    // so if we're building direct for 1.0.x (the only time NEED_OPENSSL_1_1 is undefined) then
+    // just omit all the code here.
+    //
+    // The method uses OpenSSL 1.0.x API, except for the fallback function SSL_CTX_config, to
+    // make the portable version easier.
+#ifdef NEED_OPENSSL_1_1
+
+    // Check to see if there's a registered default CipherString. If not, we will use our own.
+    SSL_CTX* ctx = SSL_CTX_new(TLS_method());
+
+    // SSL_get_ciphers returns a shared pointer, no need to save/free it.
+    // It gets invalidated every time we touch the configuration, so we can't ask just once, either.
+    SSL* ssl = SSL_new(ctx);
+    int defaultCount = sk_SSL_CIPHER_num(SSL_get_ciphers(ssl));
+    SSL_free(ssl);
+
+    int rv = SSL_CTX_set_cipher_list(ctx, "ALL");
+    assert(rv);
+
+    ssl = SSL_new(ctx);
+    int allCount = sk_SSL_CIPHER_num(SSL_get_ciphers(ssl));
+    SSL_free(ssl);
+
+    // This shouldn't be true, but, just in case it is.
```

Comment:
This comment makes it sound like the two values being true is an assertion failure, not a recoverable condition.

Reply:
Altered the wording.  Hopefully it's an improvement :)